In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [3]:
df=pd.read_csv("E:\ICTAK-DSA\StudentsPerformance.csv")
df

gender race/ethnicity parental level of education         lunch  \
0    female        group C            some high school  free/reduced   
1    female        group B                 high school  free/reduced   
2    female        group B            some high school  free/reduced   
3    female        group B                some college      standard   
4    female        group C                some college  free/reduced   
..      ...            ...                         ...           ...   
995    male        group E                some college      standard   
996    male        group A                some college      standard   
997    male        group D                some college      standard   
998    male        group E          associate's degree  free/reduced   
999    male        group E           bachelor's degree      standard   

    test preparation course  math score  reading score  writing score  
0                      none           0             17             10  
1                      none           8             24             23  
2                      none          18             32             28  
3                      none          11             38             32  
4                      none          22             39             33  
..                      ...         ...            ...            ...  
995               completed          99             87             81  
996               completed         100             96             86  
997               completed         100             97             99  
998               completed         100            100             93  
999               completed         100            100            100  

[1000 rows x 8 columns]

# 1. Find out how many males and females participated in the test

In [4]:
df['gender'].value_counts()

female    518
male      482
Name: gender, dtype: int64

>>>The number of Females students  participated in the test is 518 and Males students participation is 482.

# 2. What do you think about the students parental level of education?

In [6]:
print(df['parental level of education'].unique())

['some high school' 'high school' 'some college' "associate's degree"
 "bachelor's degree" "master's degree"]


In [7]:
df['parental level of education'].value_counts()

some college          226
associate's degree    222
high school           196
some high school      179
bachelor's degree     118
master's degree        59
Name: parental level of education, dtype: int64

In [9]:
top_parental_level = df['parental level of education'].value_counts()
top_parental_level.nlargest(1)


some college    226
Name: parental level of education, dtype: int64

In [11]:
low_parental_level = df['parental level of education'].value_counts() 
low_parental_level.nsmallest(1)

master's degree    59
Name: parental level of education, dtype: int64

>>> This dataset contains 6 level of parental education. They are some high school, high school, some college, associates degree, bachelors degree, masters degree.
Highest qualification is masters degree and it is attended by less people compared to other. And highest number done parental level of educstion in some college.

# 3. Who scores the most on average for math, reading and writing based on :

# ● Gender

In [13]:
df.groupby('gender')[['math score','reading score','writing score']].mean()

math score  reading score  writing score
gender                                          
female   63.376448      72.590734      72.467181
male     68.821577      65.545643      63.446058

>> Scoring Variation For Math, Reading and Writing based on Gender is : Female students are scored good mark in reading and writing category , while male students scored well in math.

# ● Test preparation course

In [14]:
df.groupby('test preparation course')[['math score','reading score','writing score']].mean()

math score  reading score  writing score
test preparation course                                          
completed                  69.96648      74.175978      74.684358
none                       63.78972      66.417445      64.457944

>> Scoring Variation For Math, Reading and Writing based on Test Preparation Course is : More percent of students are completed the course. On maths 69% of students completed the course , for reading and writing 74% of students completed the course.

# 4. What do you think about the scoring variation for math, reading and writing based on:


# ● Gender

In [15]:
df.groupby('gender')[['math score','reading score','writing score']].std()

math score  reading score  writing score
gender                                          
female   16.029928      14.411018      14.844842
male     14.556411      14.149594      14.227225

In [16]:
df.groupby('gender')[['math score','reading score','writing score']].var()

math score  reading score  writing score
gender                                          
female  256.958593     207.677438     220.369327
male    211.889097     200.211010     202.413924

>>Scoring Variation For Math, Reading and Writing based on Gender is : Female students are scored good mark compared to male students in each category

# ● Test preparation course

In [17]:
df.groupby('test preparation course')[['math score','reading score','writing score']].std()

math score  reading score  writing score
test preparation course                                          
completed                 14.521847      13.537572      13.236412
none                      15.705689      14.608896      15.041667

In [18]:
df.groupby('test preparation course')[['math score','reading score','writing score']].var()

math score  reading score  writing score
test preparation course                                          
completed                210.884027     183.265864     175.202612
none                     246.668662     213.419851     226.251739

>>Scoring Variation For Math, Reading and Writing based on Test Preparation Course is : More percent of students are not completed the course.
   On maths 14% of students completed the course , for reading and writing 13% of students completed the course.

# 5. The management needs your help to give bonus points to the top 25% of students based on their maths score, so how will you help the managementto achieve this.

In [64]:
bouns_point  = df.sort_values('math score',ascending=False)

In [65]:
bouns_point .quantile(0.75)['math score']

77.0

In [66]:
bouns_point .quantile(1.0)['math score']

100.0

In [69]:
bouns_point=bouns_point[bouns_point['math score'] >= 77.0]
bouns_point.reset_index()

index  gender race/ethnicity parental level of education         lunch  \
0      999    male        group E           bachelor's degree      standard   
1      996    male        group A                some college      standard   
2      515  female        group E                some college      standard   
3      517  female        group E          associate's degree      standard   
4      516  female        group E           bachelor's degree      standard   
..     ...     ...            ...                         ...           ...   
250    851    male        group D                some college      standard   
251    849    male        group A           bachelor's degree      standard   
252    850    male        group C          associate's degree  free/reduced   
253    421  female        group C                some college  free/reduced   
254    414  female        group A                 high school  free/reduced   

    test preparation course  math score  reading score  writing score  
0                 completed         100            100            100  
1                 completed         100             96             86  
2                      none         100             92             97  
3                      none         100            100            100  
4                      none         100            100            100  
..                      ...         ...            ...            ...  
250               completed          77             62             62  
251                    none          77             67             68  
252                    none          77             67             64  
253                    none          77             90             91  
254               completed          77             88             85  

[255 rows x 9 columns]

In [70]:
bouns_point['math score'].value_counts()

77     24
79     22
81     22
82     18
80     17
87     16
78     15
88     15
85     14
84     11
91      9
90      9
86      8
83      8
94      7
100     7
92      6
89      6
97      6
93      4
99      3
96      3
98      3
95      2
Name: math score, dtype: int64

In [71]:
bouns_point=bouns_point.groupby('gender')['gender'].count()
print('Number of female students getting Bonus Points :',bonus_gender[0])
print('Number of male students getting Bonus Points :',bonus_gender[1])

Number of female students getting Bonus Points : 102
Number of male students getting Bonus Points : 148


>>> The students who scored 77 or greaterthan in maths are eligible for bonus point provided by management. The total number of students eligible for bonus point is 225.
 There are 7 students who scored 100 marks in Maths.